File: Backend/scheduling/src/main/java/com/draveterinaria/cl/scheduling/model/Servicio.java
package com.draveterinaria.cl.scheduling.model;
import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.Date;
import java.util.List;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
@Table(name = "SERVICIO")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Servicio {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "service_seq")
    @SequenceGenerator(name = "service_seq", sequenceName = "seq_servicio", allocationSize = 1)
    @Column(name = "ID_SERVICIO")
    private Long idServicio;

    // Relación con MASCOTA
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_MASCOTA", nullable = false)
    private Mascota mascota;


    @Column(name = "FECHA")
    private LocalDateTime fecha;

    @Column(name = "COSTO", precision = 10, scale = 2, nullable = false)
    private BigDecimal costo;

    @Column(name = "ID_AGENDA")
    private Long idAgenda;


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_SUBTIPO", nullable = false) // Usa el nombre de columna que creaste en SQL
    private SubtipoServicio subtipo;

    @Column(name = "ID_FACTURA")
    private Long idFactura;

    @Column(name = "ESTADO_PAGO", length = 20)
    private String estadoPago = "PENDIENTE";
}

File: app/src/main/java/com/example/draveterinaria/data/model/Servicio.kt
package com.example.draveterinaria.data.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDateTime


@Entity
@Table(name = "SERVICIO")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class Servicio(

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "service_seq")
    @SequenceGenerator(name = "service_seq", sequenceName = "seq_servicio", allocationSize = 1)
    @Column(name = "ID_SERVICIO")
    var idServicio: Long? = null,
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_MASCOTA", nullable = false)
    var mascota: Mascota,

    @Column(name = "FECHA")
    var fecha: LocalDateTime,

    @Column(name = "COSTO", precision = 10, scale = 2, nullable = false)
    var costo: BigDecimal,

    @Column(name = "ID_AGENDA")
    var idAgenda: Long? = null,


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_SUBTIPO", nullable = false)
    var subtipo: SubtipoServicio,

    @Column(name = "ID_FACTURA")
    var idFactura: Long? = null,

    @Column(name = "ESTADO_PAGO", length = 20)
    var estadoPago: String = "PENDIENTE"
)